src/index.ts

```diff
@@ -1,17 +1,14 @@
 import shallowEqualArrays, { validArrayValue } from "./arrays";
 import shallowEqualObjects, { validObjectValue } from "./objects";
 
-type Comparable<T> = Record<string, T> | T[] | null | undefined;
+type Comparable = Record<string, any> | any[] | null | undefined;
 
-function shallowEqual<T extends Comparable<T>>(a: T, b: T): boolean {
+function shallowEqual<T extends Comparable>(a: T, b: T): boolean {
   if (Array.isArray(a) || Array.isArray(b)) {
```

Comment:
Should it be "&&"?

Reply:
You're right, about the philosophy of it (i.e. I made a thinking error or this); but I lucked out in being accidentally right, in that, in our implementation, you actually do want `||`. I've got an update with a test case to show why.

Reply:
(I've changed the latest implementation so that I'm not accidentally right -- but instead follow the spirit of what you're saying, which I think is far more correct)